yspryn/hw6/reduce_sum.py

```diff
@@ -0,0 +1,18 @@
+def reduce(*args, key: callable, default):
+    if args:
+        res = None
+        for i in args:
+            if res is None:
+                res = i
+            else:
+                res = key(res, i)
+        return res
+    else:
```

Comment:
try to avoid nesting in line 2,

first check if args empty and return default, after that process main function logic